src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -408,10 +415,30 @@ internal static void ConfigureApplicationInsights(HostBuilderContext context, IL
             // Initializing AppInsights services during placeholder mode as well to avoid the cost of JITting these objects during specialization
             if (!string.IsNullOrEmpty(appInsightsInstrumentationKey) || !string.IsNullOrEmpty(appInsightsConnectionString) || SystemEnvironment.Instance.IsPlaceholderModeEnabled())
             {
+                // Initialize ScriptTelemetryInitializer before any other telemetry initializers.
+                // This will allow HostInstanceId to be removed as part of MetricsCustomDimensionOptimization.
+                builder.Services.AddSingleton<ITelemetryInitializer, ScriptTelemetryInitializer>();
                 builder.AddApplicationInsightsWebJobs(o =>
                 {
                     o.InstrumentationKey = appInsightsInstrumentationKey;
                     o.ConnectionString = appInsightsConnectionString;
+
+                    if (!string.IsNullOrEmpty(context.Configuration[EnvironmentSettingNames.AppInsightsEventListenerLogLevel]))
```

Comment:
small nit for these double indexes into IConfiguration. Can use pattern matching:

``` CSHarp
if (context.Configuration[<Key>] is string value && value is not "")
{
    // use 'value' here
}
```